include/openssl/engine.h

```diff
@@ -798,9 +798,17 @@ typedef int (*dynamic_bind_engine) (ENGINE *e, const char *id,
         OPENSSL_EXPORT \
         int bind_engine(ENGINE *e, const char *id, const dynamic_fns *fns) { \
             if (ENGINE_get_static_state() == fns->static_state) goto skip_cbs; \
+            if (getenv("OPENSSL_engine_libcrypto_override") == NULL) { \
```

Comment:
Every engine will hits this if the env is not present

Reply:
Only the engines that don't share the exact same libcrypto as the loading application in process space.